src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  alive() {
+    Animal.alive.filter((creature) => creature.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
+      animal.health -= 50;
+      Animal.alive = Animal.alive.filter((el) => el.health > 0);
```

Comment:
The logic for removing dead animals from the `Animal.alive` array is only present in the `Carnivore.bite` method. This does not cover other scenarios where an animal's health might reach zero. Consider implementing a more centralized method to handle the removal of dead animals, which can be called whenever an animal's health changes.